src/main/java/org/folio/circulation/domain/notice/TemplateContextUtil.java

```diff
@@ -412,13 +415,26 @@ public UserContext withAddressProperties(JsonObject address) {
           .with(UserContext.CITY, address.getString("city", null))
           .with(UserContext.REGION, address.getString("region", null))
           .with(UserContext.POSTAL_CODE, address.getString("postalCode", null))
-          .with(UserContext.COUNTRY_ID, address.getString("countryId", null))
+          .with(UserContext.COUNTRY_ID, getCountryNameByCodeIgnoreCase(address.getString(COUNTRY_ID, null)))
           .with(UserContext.ADDRESS_TYPE_NAME, address.getString("addressTypeName", null));
       } else {
         return this;
       }
     }
 
+    public String getCountryNameByCodeIgnoreCase(String code) {
+      if (StringUtils.isEmpty(code)) {
+        return null;
+      }
+      if (Stream.of(Locale.getISOCountries()).toList().contains(code)) {
+        return new Locale("",code).getDisplayName();
+      }
+      else {
+        log.info("getCountryNameByCodeIgnoreCase:: Invalid country code {}", code);
+        throw new IllegalArgumentException("Not a valid country code to determine the country name.");
+      }
+    }
+
```

Comment:
```
    public String getCountryNameByCodeIgnoreCase(String code) {
      if (StringUtils.isEmpty(code)) {
        return null;
      }

      if (!Stream.of(Locale.getISOCountries()).toList().contains(code)) {
        log.error("getCountryNameByCodeIgnoreCase:: Invalid country code {}", code);
        throw new IllegalArgumentException("Not a valid country code to determine the country name.");
      }

      return new Locale("",code).getDisplayName();
    }
```
might be nicer. Also probably better to `log.error` there. Maybe also add a test for the exception just to make sure it's behaving the way we want with an `assertThrows`.

Also do we want to allow an empty code? Maybe the UI allows this but just checking.

Reply:
Done